src/regex.js

```diff
@@ -20,30 +21,30 @@ function toRegex(root) {
   for (let i = 0; i < states.length; i++) {
     let a = states[i];
     if (a.accepting) {
-      B[i] = new Literal('');
+      B[i] = new Literal('', flags);
     }
 
     A[i] = [];
     for (let [t, s] of a.transitions) {
       let j = states.indexOf(s);
-      A[i][j] = A[i][j] ? union(A[i][j], new Literal(t)) : new Literal(t);
+      A[i][j] = A[i][j] ? union(A[i][j], new Literal(t, flags), flags) : new Literal(t, flags);
     }
   }
 
   // Solve the of equations
   for (let n = states.length - 1; n >= 0; n--) {
     if (A[n][n] != null) {
-      B[n] = concat(star(A[n][n]), B[n]);
+      B[n] = concat(star(A[n][n]), B[n], flags);
       for (let j = 0; j < n; j++) {
-        A[n][j] = concat(star(A[n][n]), A[n][j]);
+        A[n][j] = concat(star(A[n][n]), A[n][j], flags);
       }
     }
 
     for (let i = 0; i < n; i++) {
       if (A[i][n] != null) {
-        B[i] = union(B[i], concat(A[i][n], B[n]));
+        B[i] = union(B[i], concat(A[i][n], B[n], flags), flags);
         for (let j = 0; j < n; j++) {
-          A[i][j] = union(A[i][j], concat(A[i][n], A[n][j]));
+          A[i][j] = union(A[i][j], concat(A[i][n], A[n][j], flags), flags);
```

Comment:
Hmm, will that then pass it down into other AST nodes’ `toString`s? I found the code around that quite hard to follow.